internal/nix/install.go

```diff
@@ -39,7 +40,14 @@ func Install(writer io.Writer, daemon *bool) error {
 	defer r.Close()
 
 	installScript := "curl -L https://releases.nixos.org/nix/nix-2.24.7/install | sh -s"
-	if daemon != nil {
+	if featureflag.UseDetSysInstaller.Enabled() {
+		// Should we pin version? Or just trust detsys
+		installScript = "curl --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/nix | sh -s -- install"
+		if isLinuxWithoutSystemd() {
+			installScript += " linux --init none"
```

Comment:
I presume this is to manage the nix daemon.
How does the nix daemon get started without this?

Reply:
My understanding from the DetSys [docs](https://github.com/DeterminateSystems/nix-installer) is that this makes it root only but will still work. 

There are a few alternatives to this:

* Warn and give user options (e.g. install systemd or use new `--root-only`)
* Warn and install anyway

Reply:
Seems like they'll then need to run `sudo` to run devbox. I think emitting a warning (with a suggestion to install systemd and re-using the detsys installer manually) and continuing to install seems okay to minimize friction.  WDYT?

Reply:
Not quite, user needs sudo only if not root, so at least some cases (e.g. ubuntu dockerfile) work pretty well as is.

That said, printing a message is very low friction. I'm not 100% sure re-running detsys installation would fix it, I need to experiment a bit more.